apps/icelandic-names/backend/src/app/modules/icelandic-name/icelandic-name.controller.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Delete,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+} from '@nestjs/common'
+import { ApiNoContentResponse, ApiOkResponse, ApiTags } from '@nestjs/swagger'
+
+import { IcelandicNameService } from './icelandic-name.service'
+import { IcelandicName } from './icelandic-name.model'
+import { UpdateIcelandicNameBody, CreateIcelandicNameBody } from './dto'
+
+@Controller('api')
+@ApiTags('icelandic-names')
+export class IcelandicNameController {
+  constructor(private readonly icelandicNameService: IcelandicNameService) {}
+
+  @Get('icelandic-names/all')
+  @ApiOkResponse({
+    type: IcelandicName,
+    isArray: true,
+    description: 'Gets all icelandic names',
+  })
+  getAll(): Promise<IcelandicName[]> {
+    return this.icelandicNameService.getAll()
+  }
+
+  @Get('icelandic-names/:initialLetter')
+  @ApiOkResponse({
+    type: IcelandicName,
+    isArray: true,
+    description: 'Gets all icelandic names by initial letter',
+  })
+  getByInitialLetter(
+    @Param('initialLetter') initialLetter: string,
+  ): Promise<IcelandicName[]> {
+    return this.icelandicNameService.getByInitialLetter(initialLetter)
+  }
+
+  @Post('icelandic-names/:id/update')
```

Comment:
PUT should be used for updates.
```suggestion
  @Put('icelandic-names/:id')
```